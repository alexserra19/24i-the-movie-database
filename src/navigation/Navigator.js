import * as React from 'react';
import { createStackNavigator } from '@react-navigation/stack';

import HomeScreen from '../pages/HomeScreen'
import DetailsScreen from '../pages/DetailsScreen'


Stack = createStackNavigator();

function StackNavigator() {
  return (
    <Stack.Navigator
      initialRouteName="Home"
    >
      <Stack.Screen
        name="Home"
        component={HomeScreen}
        options={{
          headerShown: false,
          headerTitle: 'Home',
          gestureEnabled: false,
        }}
      />
      <Stack.Screen
        name="Details"
        component={DetailsScreen}
        options={{
          headerShown: false,
          headerTitle: 'Details',
          gestureEnabled: false,
        }}
      />
    </Stack.Navigator>
  )
}

export default StackNavigator;

