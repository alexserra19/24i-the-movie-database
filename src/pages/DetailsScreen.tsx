import React, { useEffect, useState } from 'react';
import { View, StyleSheet, SafeAreaView, TouchableOpacity, Text, TextInput, Platform, Alert, Image } from 'react-native';
import AppConstants from '../utils/AppConstants';
import { normalize, withBadge } from 'react-native-elements';
import { useDispatch, useSelector } from "react-redux";
import { IsLoadingHoc } from "../components/HOCS/IsLoadingHOC";
import { HeaderBar } from "../components/shared/HeaderBar"
import * as RootNavigation from "../navigation/RootNavigation";

interface IDetailsScreenProps {
    navigation: any;
    setLoading: Function
}

const DetailsScreen = (props: IDetailsScreenProps) => {

    const media = useSelector((store: any) => store.mediaReducer.selectedMedia);

    const dispatch = useDispatch()


    useEffect(() => {

    }, []);

    return (
        <View style={styles.container}>
            <HeaderBar
                title={'Details'}
                onPress={() => RootNavigation.goBack()}
                containerStyle={styles.headerBar}
            />
            <View style={styles.imageContainer}>
                <Image
                    style={{ width: '100%', height: '100%', resizeMode: 'stretch' }}
                    source={{ uri: media.image }}
                />
            </View>
        </View>
    );
}

export default IsLoadingHoc(DetailsScreen);


const styles = StyleSheet.create({
    container: {
        backgroundColor: AppConstants.colors.principalColor,
        flex: 1,
        height: '100%',
        width: '100%',
    },
    imageContainer: {
        height: normalize(400),
        width: '100%',
        position: 'absolute',
        left: 0,
        right: 0,
    },
    headerBar: {
        position: 'absolute',
        top: normalize(30),
        zIndex: 3,
        alignSelf: 'center',
        padding: normalize(10)
    }
});
