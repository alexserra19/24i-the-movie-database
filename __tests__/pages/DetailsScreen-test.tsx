import 'react-native';
import React from 'react';
import DetailsScreen from '../../src/pages/DetailsScreen';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'
import { Store, AnyAction } from 'redux';
import { fireEvent, render } from '@testing-library/react-native'
import { Dimensions } from 'react-native';
import * as RootNavigation from "../../src/navigation/RootNavigation";

jest.useFakeTimers()
jest.mock('../../src/navigation/RootNavigation')

describe('DetailsScreen', () => {

  let store: Store<any, AnyAction>
  const mockStore = configureStore()

  const initialState = {
    mediaReducer: {
      selectedMedia: {}
    }
  }

  beforeEach(() => {
    store = mockStore(initialState)
  });

  const mockProps = {
    navigation: {},
    setLoading: jest.fn()
  }

  it('should match the snapshot', () => {
    const wrapper = render(
      <Provider store={store}>
        <DetailsScreen {...mockProps} />
      </Provider>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  })

  it('useEffect', () => {
    jest.spyOn(Dimensions, 'addEventListener')
    const wrapper = render(
      <Provider store={store}>
        <DetailsScreen {...mockProps} />
      </Provider>
    );
    expect(Dimensions.addEventListener).toBeCalled()
  })

  it('When Press Arrow Back from HeaderBar we should navigate back', () => {
    const wrapper = render(
      <Provider store={store}>
        <DetailsScreen {...mockProps} />
      </Provider>
    );

    jest.spyOn(RootNavigation, 'goBack')
    const button = wrapper.getByTestId('headerBarButton')
    fireEvent.press(button)

    expect(RootNavigation.goBack).toBeCalled()
  })

});
