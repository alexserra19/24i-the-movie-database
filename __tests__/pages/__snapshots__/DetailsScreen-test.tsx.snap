// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen should match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffff",
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 64.28571428571428,
          "padding": 12.857142857142858,
          "width": "100%",
        },
        Object {
          "alignSelf": "center",
          "padding": 12.857142857142858,
          "position": "absolute",
          "top": 38.57142857142857,
          "zIndex": 3,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="headerBarButton"
      >
        <Text>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#ffff",
                  "fontSize": 38.57142857142857,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 4,
          "paddingRight": 25.714285714285715,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#ffff",
              "fontSize": 25.714285714285715,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Details
      </Text>
    </View>
  </View>
  <RCTScrollView
    scrollEnabled={true}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "left": 0,
              "right": 0,
              "width": "100%",
            },
            Object {
              "height": 514.2857142857142,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/images/no-image.jpeg",
            }
          }
          style={
            Object {
              "height": "100%",
              "resizeMode": "stretch",
              "width": "100%",
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#409ff7",
              "borderRadius": 90,
              "bottom": -38.57142857142857,
              "elevation": 9,
              "height": 90,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 10,
              "position": "absolute",
              "right": 25.714285714285715,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.32,
              "shadowRadius": 5.46,
              "width": 90,
            }
          }
        >
          <Text>
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#ffff",
                    "fontSize": 38.57142857142857,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </Text>
        </View>
      </View>
      <RCTSafeAreaView
        emulateUnlessSupported={true}
      >
        <View
          style={
            Object {
              "marginTop": 25.714285714285715,
              "padding": 12.857142857142858,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 32.14285714285714,
                "marginBottom": 12.857142857142858,
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 23.142857142857142,
                "marginBottom": 25.714285714285715,
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 21.857142857142858,
                "lineHeight": 28,
                "textAlign": "justify",
              }
            }
          />
        </View>
      </RCTSafeAreaView>
    </View>
  </RCTScrollView>
</View>
`;
